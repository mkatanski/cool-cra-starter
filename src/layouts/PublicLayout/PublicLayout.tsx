import React from "react";
import LayoutBase, { LayoutBaseProps } from "layouts/LayoutBase";

const PublicLayout: React.FC<LayoutBaseProps> = ({
  children,
  ...rest
}: LayoutBaseProps) => {
  return (
    <>
      <LayoutBase {...rest}>{children}</LayoutBase>
    </>
  );
};

export default PublicLayout;
