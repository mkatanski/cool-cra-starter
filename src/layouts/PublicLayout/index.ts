import PublicLayout from "./PublicLayout";
export * from "./PublicLayout";

export default PublicLayout;
