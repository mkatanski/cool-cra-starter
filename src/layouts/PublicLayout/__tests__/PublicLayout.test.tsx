/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";

import PublicLayout from "../PublicLayout";

jest.mock(
  "layouts/LayoutBase",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="LayoutBase" {...rest}>
      {children}
    </div>
  )
);

describe("layouts", () => {
  describe("PublicLayout", () => {
    it("should render PublicLayout component", async () => {
      const tree = TestRenderer.create(<PublicLayout />);
      expect(tree).toMatchSnapshot();
    });

    it("should render PublicLayout with children", () => {
      const tree = TestRenderer.create(
        <PublicLayout>Some children</PublicLayout>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
