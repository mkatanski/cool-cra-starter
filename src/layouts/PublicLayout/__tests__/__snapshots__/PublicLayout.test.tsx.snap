// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts PublicLayout should render PublicLayout component 1`] = `
Array [
  .c0 {
  position: fixed;
  right: 0;
  bottom: 0;
  z-index: 1;
  padding: 6px 18px 10px 6px;
}

<aside
    className="c0"
  >
    <button
      id="LangSwitcherButton"
      isActive={false}
      label="English"
      value="en"
    />
    <button
      id="LangSwitcherButton"
      isActive={false}
      label="Français"
      value="fr"
    />
  </aside>,
  <div
    id="LayoutBase"
  />,
]
`;

exports[`layouts PublicLayout should render PublicLayout with children 1`] = `
Array [
  .c0 {
  position: fixed;
  right: 0;
  bottom: 0;
  z-index: 1;
  padding: 6px 18px 10px 6px;
}

<aside
    className="c0"
  >
    <button
      id="LangSwitcherButton"
      isActive={false}
      label="English"
      value="en"
    />
    <button
      id="LangSwitcherButton"
      isActive={false}
      label="Français"
      value="fr"
    />
  </aside>,
  <div
    id="LayoutBase"
  >
    Some children
  </div>,
]
`;
